iphone/Classes/TopTiModule.h

```diff
@@ -1,13 +1,31 @@
 /**
  * Appcelerator Titanium Mobile
- * Copyright (c) 2009-2010 by Appcelerator, Inc. All Rights Reserved.
+ * Copyright (c) 2009-2018 by Appcelerator, Inc. All Rights Reserved.
  * Licensed under the terms of the Apache Public License
  * Please see the LICENSE included with this distribution for details.
  */
+#import <JavaScriptCore/JavaScriptCore.h>
 
-#import "TiModule.h"
+@protocol TiExports <JSExport>
 
-@interface TopTiModule : TiModule {
-}
+@property (readonly, nonatomic) NSString *apiName;
+@property (readonly, nonatomic) NSString *buildDate;
+@property (readonly, nonatomic) NSString *buildHash;
+@property (readwrite, nonatomic) NSString *userAgent;
+@property (readonly, nonatomic) NSString *version;
 
+- (JSValue *)createBuffer:(NSDictionary *)arg;
+
+// Accessors from JS. This is pretty redundant, Maybe introduce macros for this common case?
+// (i.e. readonly properties also generate getter methods, readwrite generator getter/setters)
+- (NSString *)getBuildDate;
```

Comment:
in the current implementation, a `@property` in the header can be written as the following inside the implemention:
```objc
- (NSString *)buildDate
{
  return @"____";
}
```
Important: This allows it to be used both as a get-property and getter-method. So no manual `getX` method declaration in the header required, only if we explicitly want one, like if the getter has arguments.

Reply:
Yeah and this difference was what raised the removal of getX/setX accessor methods to my mind when doing this PR. I know that our proxy hack will fall back to mangling `getX` to look for an `x` property (or more specifically just a `x` method with no args). 

With the new API, that doesn't happen. It's not a JS idiom, so they didn't build it in. So with the new binding API we need to make the getter/setter methods explicitly and just have them delegate over. (which would be ok if we could write macros to do it for us, but macros aren't flexible enough to camelcase a property name)